inc/add_site_ui/namespace.php

```diff
@@ -0,0 +1,291 @@
+						<td>
+							<?php
+							// Network default.
+							$lang = get_site_option( 'WPLANG' );
+
+							// Use English if the default isn't available.
+							if ( ! in_array( $lang, $languages ) ) {
+								$lang = '';
+							}
+
+							wp_dropdown_languages(
+								[
+									'name'                        => 'language',
+									'id'                          => 'site-language',
+									'selected'                    => $lang,
+									'languages'                   => $languages,
+									'translations'                => [],
+									'show_available_translations' => false,
+								]
+							);
+							?>
+						</td>
+					</tr>
+				<?php endif; // Languages. ?>
+				<tr class="form-field">
+					<th scope="row"><?php _e( 'Public', 'altis' ); ?></th>
+					<td>
+						<label>
+							<input name="public" type="checkbox" id="public" value="1" aria-describedby="public-desc" checked />
+							<strong><?php _e( 'Should this site be public?', 'altis' ); ?> </strong>
+							<span class="input-description" id="public-desc"><?php _e( 'Uncheck to require login.', 'altis' ) ?></span>
+						</label>
+					</td>
+				</tr>
+			</table>
+
+			<?php
+			/**
+			 * Fires at the end of the new site form in network admin.
+			 *
+			 * @since 4.5.0
+			 */
+			do_action( 'network_site_new_form' );
+
+			submit_button( __( 'Add Site' ), 'primary', 'add-site' );
+			?>
+		</form>
+	</div>
+	<?php
+
+	require ABSPATH . 'wp-admin/admin-footer.php';
+
+	// Exit before we attempt to render WordPress' about page.
+	exit;
+}
+
+/**
+ * Handler function for the Add Site form submission.
+ */
+function add_site_form_handler() {
+	if ( ! isset( $_POST['_wpnonce'] ) || ! wp_verify_nonce( $_POST['_wpnonce'], 'altis-add-site' ) ) {
+		return;
+	}
+
+	$site_type_valid_values = [
+		'site-subdomain',
+		'site-subdirectory',
+		'site-custom-domain',
+	];
+
+	$site_type_value = sanitize_text_field( $_POST['domain-type'] );
+
+	if ( in_array( $site_type_value, $site_type_valid_values ) ) {
+		$site_type = $site_type_value;
+	} else {
+		$site_type = 'site-subdomain';
+	}
+
+	$network_url = wp_parse_url( network_site_url() );
+	$network_url = $network_url['host'];
+	$form_url    = sanitize_text_field( $_POST['url'] );
+	$url         = $form_url;
+	$title       = sanitize_text_field( $_POST['title'] );
+	$language    = sanitize_text_field( $_POST['language'] ) ?? '';
+	$public      = sanitize_text_field( $_POST['public'] );
+	$domain      = '';
+	$path        = '';
+
+	if ( ( strpos( $url, $network_url ) && strpos( $url, 'http' ) !== 0 )
+		|| ( 'site-custom-domain' === $site_type && strpos( $url, 'http' ) !== 0 ) ) {
+		$url = 'https://' . $url;
+	}
+
+	if ( wp_http_validate_url( $url ) ) {
+		$url_array = wp_parse_url( $url );
+		$domain    = str_replace( $network_url, '', $url_array['host'] );
+		$domain    = trim( $domain, '.' );
+		$path      = trim( $url_array['path'], '/' );
+	} else {
+		$domain = trim( $url, '.' );
+		$path   = trim( $url, '/' );
```

Comment:
What was the reasoning for using `$url` to set both of these values? I think there's some mixed logic here because `wp_http_validate_url()` will not always be true for a custom domain, there are many reasons why it can fail.

We should ensure that these values are only set under the right `$site_type` contexts.